source/common/matcher/solo_matcher.cc

```diff
@@ -118,10 +118,12 @@ class PathMatcherImpl : public BaseMatcherImpl {
 
 class CompiledStdMatcher : public Regex::CompiledMatcher {
 public:
-  CompiledStdMatcher(std::regex &&regex) : regex_(std::move(regex)) {}
+  CompiledStdMatcher(const std::string &pattern) :
```

Comment:
The better pattern for this is:
```suggestion
  CompiledStdMatcher(const std::string pattern) :
  pattern_(std::move(pattern)),
```
This way, the caller can actually move the string into here.

Reply:
Should be fixed in 31208c53037892a0e894a0d91573d7719f474bb5 but would appreciate if you could take a second look. I'm still a bit shaky on exactly how move semantics work.

I also thought about using `std::move` when calling the `CompiledStdMatcher` constructor too:

```
diff --git a/source/common/matcher/solo_matcher.cc b/source/common/matcher/solo_matcher.cc
index 48d27b6..7165efe 100644
--- a/source/common/matcher/solo_matcher.cc
+++ b/source/common/matcher/solo_matcher.cc
@@ -153,7 +153,7 @@ private:
 class StdRegexEngine : public Regex::Engine {
 public:
   absl::StatusOr<Regex::CompiledMatcherPtr> matcher(const std::string &regex_pattern) const override {
-    return std::make_unique<CompiledStdMatcher>(std::move(regex_pattern));
+    return std::make_unique<CompiledStdMatcher>(regex_pattern);
   }
 };
```

but then i reverted it because i wasn't sure if the calling function was still using the string, so i thought it might be unsafe. let me know what you think - thanks!

Reply:
correct, you cannot use move here because it's a reference, you don't know where that string is and if the user still needs it. Like my suggestion for the CompiledStdMatcher, if you want to use move here, you can just change that regex_pattern to be pass by value, then it's save to move. You are making a copy anyway, the key is to make a copy only once, whether the compiler does it or you manually do it. When the compiler does it, it can optimize where it can.

Reply:
wait, you cannot change the function signature here because is "override", so what you have now is correct.